### DISCUSSION TITLE
Python Async

### DISCUSSION THREAD

**Discussion by tybreakyr** [ravendb/ravendb#17867] [Q&A]
First I'd like to say that the zero-to-RavenDB doc is a great read and just having available for users to read sets this product well above the competition. I love the idea behind the product and hope to incorporate it in a personal project soon.

On to the pointy end of the stick... 
I'm wonder if the python client does/will support python async functionality. I have a project that uses strawberry-graphql and FastApi and I think RavenDB could make a great datasource for such an application but performance would be hindered and complexity increased without native async support.
I wasn't able to find any documentation mentioning how to use the python client asynchronously so if it's already built-in some hints on how to use effectively would be appreciated.

**Comment by poissoncorp** [comment]
Hello @tybreakyr!
RavenDB Python client currently doesn't expose async API, but it might become a thing in the future. The client is now being [continuously developed](https://github.com/ravendb/ravendb-python-client/pulls) to handle new features. We plan to support the full set of RavenDB features, which is the state we're just about to achieve. After that, we may plan to implement async API in the future. 

I've opened an issue, so you can track the changes.
https://issues.hibernatingrhinos.com/issue/RDBC-773/Implement-and-support-async-API
